src/main/java/dk/kb/present/PresentFacade.java

```diff
@@ -134,23 +134,10 @@ public static StreamingOutput getRecords(
             HttpServletResponse httpServletResponse, String originID, Long mTime, Long maxRecords, String format,
             Function<List<String>, List<String>> accessChecker) {
         DSOrigin origin = originHandler.getOrigin(originID);
-        if (origin == null) {
-            throw new InvalidArgumentServiceException(String.format(
-                    Locale.ROOT, "The origin '%s' was unknown. Known origins are %s",
-                    originID,
-                    originHandler.getOrigins().stream().map(DSOrigin::getId).collect(Collectors.toList())));
-        }
-
-        // Batch-oriented filter that only passed records that are allowed
-        Function<List<DsRecordDto>, Stream<DsRecordDto>> accessFilter = records -> {
-            List<String> allIDs = records.stream().map(DsRecordDto::getId).collect(Collectors.toList());
-            Set<String> allowedIDs = new HashSet<>(accessChecker.apply(allIDs));
-            return records.size() == allowedIDs.size() ?
-                    records.stream() : // No need for checking as all IDs passed
-                    records.stream().filter(record -> allowedIDs.contains(record.getId()));
-        };
+        Function<List<DsRecordDto>, Stream<DsRecordDto>> accessFilter = validateAccessForRecords(originID, accessChecker, origin);
 
-        // enum:  ['JSON-LD', 'JSON-LD-Lines', 'MODS', 'SolrJSON', "StorageRecord"]
+        // Maybe re-write this switch to use an actual ENUM?
+        // enum:  ['JSON-LD', 'JSON-LD-Lines', 'MODS', 'SolrJSON']
```

Comment:
Oh yes. Except that I'm fairly sure that `-` is nor allowed in enum IDs, so we need to make an enum where the String representation differs from the IDs. Not a problem, just a note.

Reply:
We'll do that at a later time.